src/resources/views/crud/filters/select2_ajax.blade.php

```diff
@@ -115,12 +115,9 @@ class="nav-item dropdown {{ Request::get($filter->name)?'active':'' }}">
 					var ajax_table = $('#crudTable').DataTable();
 					var current_url = ajax_table.ajax.url();
 					var new_url = addOrUpdateUriParameter(current_url, parameter, val);
-					if (val_text) {
-	                    new_url = addOrUpdateUriParameter(new_url, parameter + '_text', val_text);
-					}
+					new_url = addOrUpdateUriParameter(new_url, parameter + '_text', val_text);
```

Comment:
By removing the validation to check if `val_text` exists, this function will run always, so when the value is removed it will clear the `filter_text` from URL.